routers/repo/branch.go

```diff
@@ -148,10 +148,13 @@ func loadBranches(ctx *context.Context) []*Branch {
 			return nil
 		}
 
-		isProtected, err := ctx.Repo.Repository.IsProtectedBranch(rawBranches[i].Name, ctx.User)
-		if err != nil {
-			ctx.ServerError("IsProtectedBranch", err)
-			return nil
+		var isProtected bool
```

Comment:
isProtected maybe default `true`?

Reply:
I think it would be better to add a check before https://github.com/go-gitea/gitea/blob/19bf4ddf800dd405bddbafee7574ff4df07ed275/models/branches.go#L167 
If we are here and user is nil -> return true, nil

Reply:
This will fix this for any use case of IsProtectedBranch

Reply:
Yes, I thought so once, but the bad thing is the IsProtectedBranch function will be invoked multi-times redundantly.

Reply:
So simply do at start of func as it would be the same.
```
if doer  == nil {
  return true, nil
}
```

Reply:
OK, I updated, thanks.